src/main/java/edu/harvard/iq/dataverse/api/LocalContexts.java

```diff
@@ -0,0 +1,172 @@
+import edu.harvard.iq.dataverse.authorization.users.AuthenticatedUser;
+import edu.harvard.iq.dataverse.engine.command.DataverseRequest;
+import edu.harvard.iq.dataverse.engine.command.impl.GetDatasetCommand;
+import edu.harvard.iq.dataverse.settings.JvmSettings;
+import edu.harvard.iq.dataverse.util.json.JsonPrinter;
+import edu.harvard.iq.dataverse.util.json.JsonUtil;
+import jakarta.ejb.EJB;
+import jakarta.inject.Inject;
+import jakarta.json.Json;
+import jakarta.json.JsonObject;
+import jakarta.ws.rs.*;
+import jakarta.ws.rs.core.Context;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.container.ContainerRequestContext;
+
+@Path("localcontexts")
+public class LocalContexts extends AbstractApiBean {
+
+    protected static final Logger logger = Logger.getLogger(LocalContexts.class.getName());
+
+    @EJB
+    DatasetServiceBean datasetService;
+
+    @Inject
+    DataverseRequestServiceBean dvRequestService;
+
+    @EJB
+    PermissionServiceBean permissionService;
+
+    @GET
+    @Path("/datasets/{id}")
+    @Produces(MediaType.APPLICATION_JSON)
+    @AuthRequired
+    public Response getDatasetLocalContexts(@Context ContainerRequestContext crc, @PathParam("id") String id) {
+        try {
+            Dataset dataset = findDatasetOrDie(id);
+            DataverseRequest req = createDataverseRequest(getRequestUser(crc));
+
+            // Check if the user has edit dataset permission
+            if (!permissionService.userOn(req.getUser(), dataset).has(Permission.EditDataset)) {
+                return error(Response.Status.FORBIDDEN,
+                        "You do not have permission to query LocalContexts about this dataset.");
+            }
+
+            String localContextsUrl = JvmSettings.LOCALCONTEXTS_URL.lookupOptional().orElse(null);
+            String localContextsApiKey = JvmSettings.LOCALCONTEXTS_API_KEY.lookupOptional().orElse(null);
+
+            if (localContextsUrl == null || localContextsApiKey == null) {
+                return error(Response.Status.NOT_FOUND, "LocalContexts API configuration is missing.");
+            }
+
+            String datasetDoi = dataset.getGlobalId().asString();
+            String apiUrl = localContextsUrl + "api/v2/projects/?publication_doi=" + datasetDoi;
+            logger.info("URL used: " + apiUrl);
+            try {
+                HttpClient client = HttpClient.newHttpClient();
+                HttpRequest request = HttpRequest.newBuilder().uri(new URI(apiUrl))
+                        .header("X-Api-Key", localContextsApiKey).GET().build();
+
+                HttpResponse<String> response;
+
+                response = client.send(request, HttpResponse.BodyHandlers.ofString());
+
+                if (response.statusCode() == 200) {
+                    // Assuming the response is already in JSON format
+                    logger.info("Response from search: " + response.body());
```

Comment:
```suggestion
                    logger.fine("Response from search: " + response.body());
```

Please reduce to fine in other "happy path" places, please.